drivers/azure/azure.go

```diff
@@ -415,7 +432,7 @@ func (d *Driver) Create() error {
 	if err := c.CreateAvailabilitySetIfNotExists(ctx, d.deploymentCtx, d.ResourceGroup, d.AvailabilitySet, d.Location, d.ManagedDisks, int32(d.FaultCount), int32(d.UpdateCount)); err != nil {
 		return err
 	}
-	if err := c.CreateNetworkSecurityGroup(ctx, d.deploymentCtx, d.ResourceGroup, d.naming().NSG(), d.Location, d.deploymentCtx.FirewallRules); err != nil {
+	if err := c.CreateNetworkSecurityGroup(ctx, d.deploymentCtx, d.ResourceGroup, d.nsgResource, d.Location, d.nsgUsedInPool, d.deploymentCtx.FirewallRules); err != nil {
```

Comment:
Can we update the sig of CreateNetworkSecurityGroup to just accept `d`?

Reply:
It's cyclical, we will address later.

Reply:
Filed an issue https://github.com/rancher/rancher/issues/27189